[CI] GeoHexTilerTests testLargeShapeWithBounds failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/g66v3qvjvp5my/tests/:x-pack:plugin:spatial:test/org.elasticsearch.xpack.spatial.search.aggregations.bucket.geogrid.GeoHexTilerTests/testLargeShapeWithBounds

**Reproduction line:**
```
./gradlew ':x-pack:plugin:spatial:test' --tests "org.elasticsearch.xpack.spatial.search.aggregations.bucket.geogrid.GeoHexTilerTests.testLargeShapeWithBounds" -Dtests.seed=6AEDD755E6BE536B -Dtests.locale=en-NZ -Dtests.timezone=America/Grenada -Druntime.java=18
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.spatial.search.aggregations.bucket.geogrid.GeoHexTilerTests&tests.test=testLargeShapeWithBounds

**Failure excerpt:**
```
java.lang.AssertionError: Failed POINT (78.0 -86.42830748478096) at resolution 5 with precision 5

  at __randomizedtesting.SeedInfo.seed([6AEDD755E6BE536B:7842357C90855F2D]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.junit.Assert.assertTrue(Assert.java:41)
  at org.elasticsearch.xpack.spatial.search.aggregations.bucket.geogrid.GeoHexTilerTests.assertCorner(GeoHexTilerTests.java:131)
  at org.elasticsearch.xpack.spatial.search.aggregations.bucket.geogrid.GeoHexTilerTests.testLargeShapeWithBounds(GeoHexTilerTests.java:114)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:577)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/BoundedGeoHexGridTiler.java
+++ b/x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/BoundedGeoHexGridTiler.java
@@ -25,29 +25,33 @@ import java.io.IOException;
  * by half of the width and half of the height.
  */
 public class BoundedGeoHexGridTiler extends AbstractGeoHexGridTiler {
-    private final GeoBoundingBox inflatedBbox;
+    private final GeoBoundingBox[] inflatedBboxes;
     private final GeoBoundingBox bbox;
     private final GeoHexVisitor visitor;
-    private final int precision;
-    private static final double FACTOR = 0.06;
+    private final int resolution;
+    private static final double FACTOR = 0.5;
 
-    public BoundedGeoHexGridTiler(int precision, GeoBoundingBox bbox) {
-        super(precision);
+    public BoundedGeoHexGridTiler(int resolution, GeoBoundingBox bbox) {
+        super(resolution);
         this.bbox = bbox;
         this.visitor = new GeoHexVisitor();
-        this.precision = precision;
-        inflatedBbox = inflateBbox(precision, bbox);
+        this.resolution = resolution;
+        inflatedBboxes = new GeoBoundingBox[resolution];
+        for (int i = 0; i < resolution; i++) {
+            inflatedBboxes[i] = inflateBbox(i, bbox);
+        }
     }
 
     private static GeoBoundingBox inflateBbox(int precision, GeoBoundingBox bbox) {
         /*
         * Here is the tricky part of this approach. We need to be able to filter cells at higher precisions
-        * because they are not in bounds, but we need to make sure we don't filter too much. We use h3 bins at the given
-        * resolution to check the height ands width at that level, and we factor it depending on the precision.
+        * because they are not in bounds, but we need to make sure we don't filter too much.
+        *
+        * We use h3 bins at the given resolution to check the height ands width at that level, and we add half of it.
         *
         * The values have been tune using test GeoHexTilerTests#testLargeShapeWithBounds
         */
-        final double factor = FACTOR * (1 << precision);
+        final double factor = FACTOR;
         final Rectangle minMin = H3CartesianUtil.toBoundingBox(H3.geoToH3(bbox.bottom(), bbox.left(), precision));
         final Rectangle maxMax = H3CartesianUtil.toBoundingBox(H3.geoToH3(bbox.top(), bbox.right(), precision));
         // compute height and width at the given precision
@@ -89,15 +93,16 @@ public class BoundedGeoHexGridTiler extends AbstractGeoHexGridTiler {
     @Override
     protected long getMaxCells() {
         // TODO: Calculate correctly based on bounds
-        return UnboundedGeoHexGridTiler.calcMaxAddresses(precision);
+        return UnboundedGeoHexGridTiler.calcMaxAddresses(resolution);
     }
 
     @Override
     protected boolean h3IntersectsBounds(long h3) {
         visitor.reset(h3);
         final int resolution = H3.getResolution(h3);
-        if (resolution != precision) {
-            return cellIntersectsBounds(visitor, inflatedBbox);
+        if (resolution != this.resolution) {
+            assert resolution < this.resolution;
+            return cellIntersectsBounds(visitor, inflatedBboxes[resolution]);
         }
         return cellIntersectsBounds(visitor, bbox);
     }
@@ -106,8 +111,9 @@ public class BoundedGeoHexGridTiler extends AbstractGeoHexGridTiler {
     protected GeoRelation relateTile(GeoShapeValues.GeoShapeValue geoValue, long h3) throws IOException {
         visitor.reset(h3);
         final int resolution = H3.getResolution(h3);
-        if (resolution != precision) {
-            if (cellIntersectsBounds(visitor, inflatedBbox)) {
+        if (resolution != this.resolution) {
+            assert resolution < this.resolution;
+            if (cellIntersectsBounds(visitor, inflatedBboxes[resolution])) {
                 // close to the poles, the properties of the H3 grid are lost because of the equirectangular projection,
                 // therefore we cannot ensure that the relationship at this level make any sense in the next level.
                 // Therefore, we just return CROSSES which just mean keep recursing.

--- a/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoHexTilerTests.java
+++ b/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoHexTilerTests.java
@@ -82,13 +82,6 @@ public class GeoHexTilerTests extends GeoGridTilerTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/92541")
-    @Override
-    public void testGeoGridSetValuesBoundingBoxes_BoundedGeoShapeCellValues() throws Exception {
-        super.testGeoGridSetValuesBoundingBoxes_BoundedGeoShapeCellValues();
-    }
-
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/92540")
     public void testLargeShapeWithBounds() throws Exception {
         // We have a shape covering all space
         Rectangle shapeRectangle = new Rectangle(-180, 180, 90, -90);

